be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala

```diff
@@ -6,28 +6,31 @@ import akka.stream.scaladsl.Flow
 import ch.epfl.pop.model.network.JsonRpcRequest
 import ch.epfl.pop.model.network.method.message.Message
 import ch.epfl.pop.pubsub.AskPatternConstants
+import ch.epfl.pop.pubsub.graph.DbActor.{DbActorNAck, DbActorWriteAck}
 import ch.epfl.pop.pubsub.graph.{DbActor, ErrorCodes, GraphMessage, PipelineError}
 
-import scala.concurrent.Await
+import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 
 trait MessageHandler extends AskPatternConstants {
-  implicit lazy val dbActor: AskableActorRef = DbActor.getInstance
+  implicit final val dbActor: AskableActorRef = DbActor.getInstance
 
   val handler: Flow[GraphMessage, GraphMessage, NotUsed]
 
   def dbAskWritePropagate(rpcMessage: JsonRpcRequest): GraphMessage = {
     val paramsMessage: Option[Message] = rpcMessage.getParamsMessage
     paramsMessage match {
-      case Some(message) =>
-        // val ask = dbActor.ask(ref => DbActor.Write(rpcMessage.getParamsChannel, message, ref)).map {
-        val ask = dbActor.ask("m").map {
-          case true =>
+      case Some(message: Message) =>
+        val f: Future[GraphMessage] = (dbActor ? DbActor.Write(rpcMessage.getParamsChannel, message)).map {
+          case DbActorWriteAck =>
             // FIXME propagate
```

Comment:
propagate as in... broadcast to other clients? if so, that still really needs to be implemented.